docs/website/contents/for-developers/utxo-hd/utxo-hd.md

```diff
@@ -0,0 +1,436 @@
+this data structure holds a UTXO set which is a `Map` from
+`TxIn (EraCrypto era)` to `TxOut era`.
+
+In order to apply the different Ledger operations, there is no need for this set
+to be complete at all times, as only entries consumed by the transactions will be
+accessed. When given a block or a transaction, the Ledger code provides functions
+for getting the set of keys that would be necessary to exist in the UTXO set for
+that block or transaction to apply correctly.Taking advantage of this, the Consensus layer will modify this
+container such that it only contains the entries necessary for the Ledger rules. 
+
+### Shelley instantiation and ledger tables
+
+The `LedgerState (ShelleyBlock proto era)` data family instances are augmented
+with a new field which will hold these entries that will be extracted from and
+injected to the `NewEpochState` before calling the Ledger rules. This new field
+(which we call _ledger tables_) is a container-like structure parametrized by a
+`Key` and `Value` type families.
+
+```diff haskell
+data instance LedgerState (ShelleyBlock proto era) mk = ShelleyLedgerState {
+       shelleyLedgerTip        :: !(WithOrigin (ShelleyTip proto era))
+     , shelleyLedgerState      :: !(SL.NewEpochState era)
+     , shelleyLedgerTransition :: !ShelleyTransition
++    , shelleyLedgerTables     :: !(LedgerTables (LedgerState (ShelleyBlock proto era)) mk)
+    }
+
+data LedgerTables l mk = LedgerTables {
+    getLedgerTables :: mk (Key l) (Value l)
+}
+```
+
+For a Shelley block, these type families are mapped to the same types as above:
+
+- `Key (LedgerState (ShelleyBlock proto era)) = TxIn (EraCrypto era)`
+- `Value (LedgerState (ShelleyBlock proto era)) = TxOut era`
+
+To instantiate the `mk` type variable, some _mapkinds_ are defined:
+
+| `MapKind :: Type -> Type -> Type` | Container                     | Used  for                                                         |
+|-----------------------------------|-------------------------------|------------------------------------------------------------------|
+| `ValuesMK k v`                    | `Map k v`                     | Ledger states passed to and from the Ledger rules                |
+| `KeysMK k v`                      | `Set k`                       | Querying the disk for the values needed by a block               |
+| `DiffMK k v`                      | `Map k (Delta v)`             | Carrying the differences created by applying a block             |
+| `SeqDiffMK k v`                   | `FingerTree SlotNo (Map k v)` | Carrying the differences for the last `k` blocks |
+| `EmptyMK k v`                     | $\emptyset$                   | When not needing info about the UTxO set, or the values are inside the `NewEpochState`                         |
+| `TrackingMK k v`                  | `(Map k v, Map k (Delta v))`  | Carrying both `ValuesMK` and `DiffMK`                            |
```

Comment:
If we want to give the definitions of mapkinds, then I think we could only mention `ValuesMK`, `KeysMK`, `DiffMK` and `EmptyMK`